sdk/storage/AzureStorageBlob/Source/DataStructures/BlobDownloadOperations.swift

```diff
@@ -42,11 +42,19 @@ internal class BlobDownloadInitialOperation: ResumableTransfer {
 
     internal func queueRemainingBlocks(forTransfer transfer: BlobTransfer) {
         guard transfer.transferType == .download else { return }
-        guard let operation = transfer.operation else { return }
+        guard let opQueue = operationQueue else { return }
+        guard let downloader = transfer.downloader else { return }
+        guard downloader.blockList.count > 0 else { return }
         guard let context = transfer.managedObjectContext else { return }
 
-        var operations = [BlockOperation]()
-        guard let downloader = transfer.downloader else { return }
+        // The intialDownloadOperation marks the transfer complete, so
+        // if blocks remain, we must reset the state to inProgress.
+        transfer.state = .inProgress
+
+        var operations = [ResumableOperation]()
+        let finalOperation = BlobDownloadFinalOperation(withTransfer: transfer, queue: opQueue)
+        operations.append(finalOperation)
```

Comment:
No, because the final operation is dependent on all the other operations completing. The operations don't get added to the operations queue until all of those dependencies are defined.